src/Packagist/WebBundle/Controller/PackageController.php

```diff
@@ -142,47 +140,44 @@ public function fetchInfoAction(Request $req)
         $user = $this->getUser();
         $package->addMaintainer($user);
 
-        $response = array('status' => 'error', 'reason' => 'No data posted.');
-        if ('POST' === $req->getMethod()) {
-            $form->bind($req);
-            if ($form->isValid()) {
-                list(, $name) = explode('/', $package->getName(), 2);
-
-                $existingPackages = $this->getDoctrine()
-                    ->getRepository('PackagistWebBundle:Package')
-                    ->createQueryBuilder('p')
-                    ->where('p.name LIKE ?0')
-                    ->setParameters(array('%/'.$name))
-                    ->getQuery()
-                    ->getResult();
-
-                $similar = array();
-
-                /** @var Package $existingPackage */
-                foreach ($existingPackages as $existingPackage) {
-                    $similar[] = array(
-                        'name' => $existingPackage->getName(),
-                        'url' => $this->generateUrl('view_package', array('name' => $existingPackage->getName()), true),
-                    );
-                }
+        $form->handleRequest($req);
+        if ($form->isValid()) {
+            list(, $name) = explode('/', $package->getName(), 2);
 
-                $response = array('status' => 'success', 'name' => $package->getName(), 'similar' => $similar);
-            } else {
-                $errors = array();
-                if (count($form->getErrors())) {
-                    foreach ($form->getErrors() as $error) {
-                        $errors[] = $error->getMessageTemplate();
-                    }
+            $existingPackages = $this->getDoctrine()
+                ->getRepository('PackagistWebBundle:Package')
+                ->createQueryBuilder('p')
+                ->where('p.name LIKE ?0')
+                ->setParameters(array('%/'.$name))
+                ->getQuery()
+                ->getResult();
+
+            $similar = array();
+
+            /** @var Package $existingPackage */
+            foreach ($existingPackages as $existingPackage) {
+                $similar[] = array(
+                    'name' => $existingPackage->getName(),
+                    'url' => $this->generateUrl('view_package', array('name' => $existingPackage->getName()), true),
+                );
+            }
+
+            $response = array('status' => 'success', 'name' => $package->getName(), 'similar' => $similar);
+        } else {
```

Comment:
As per @stof's comment, I think if you make this an `} elseif ($form->isSubmitted()) {` then you can wrap up with an `} else {` in which you return `$response = array('status' => 'error', 'reason' => 'No data posted.');`.